Userland/Libraries/LibWeb/HTML/HTMLTextAreaElement.cpp

```diff
@@ -134,21 +181,46 @@ void HTMLTextAreaElement::create_shadow_tree_if_needed()
         return;
 
     auto shadow_root = heap().allocate<DOM::ShadowRoot>(realm(), document(), *this, Bindings::ShadowRootMode::Closed);
+    set_shadow_root(shadow_root);
+
     auto element = MUST(DOM::create_element(document(), HTML::TagNames::div, Namespace::HTML));
+    MUST(shadow_root->append_child(element));
+
+    m_placeholder_element = MUST(DOM::create_element(document(), HTML::TagNames::div, Namespace::HTML));
+    m_placeholder_element->set_use_pseudo_element(CSS::Selector::PseudoElement::Placeholder);
```

Comment:
this code won't compile if you rebase this branch because `Placeholder` is in `PseudoElement::Type` now.

Reply:
Good find, i will rebase